undertow/src/main/java/org/wildfly/extension/undertow/HttpListenerResourceDefinition.java

```diff
@@ -42,10 +44,14 @@
 
 /**
  * @author <a href="mailto:tomaz.cerar@redhat.com">Tomaz Cerar</a> (c) 2012 Red Hat Inc.
+ * @author Richard Achmatowicz (c) 2020 Red Hat Inc.
  */
 public class HttpListenerResourceDefinition extends ListenerResourceDefinition {
-    protected static final HttpListenerResourceDefinition INSTANCE = new HttpListenerResourceDefinition();
 
+    static final RuntimeCapability<Void> HTTP_UPGRADE_REGISTRY_CAPABILITY = RuntimeCapability.Builder.of(Capabilities.CAPABILITY_HTTP_UPGRADE_REGISTRY, true, ChannelUpgradeHandler.class)
```

Comment:
If the use of this capability outside this subsystem only needs a dependency for ordering reasons and doesn't need access to a particular type, then the capability should not provide a type, as that becomes an unneeded part of its API contract.

ServerAdd in messaging-activemq doesn't seem to need anything but a dependency; it doesn't inject.

If that's the case in other external uses, internal-to-the-subsystem wiring that might need to inject ChannelUpgradeHandler doesn't have to use the capability.

Reply:
The endpoint @rachmatowicz has in mind is this one:
https://github.com/wildfly/wildfly-core/blob/2f3e21463e446fc5aad1e5b24f4ec1222e02e9ab/remoting/subsystem/src/main/java/org/jboss/as/remoting/RemotingHttpUpgradeService.java#L131

Thus, I believe the service type should be part of the capability contract.

Reply:
OK, thanks. @fl4via should look at this overall, but in particular that point as it makes the contract a bit stronger to include it in the capability registry.

Of course the coupling already exists but when things get formalized like this that's a good time to confirm the contracts are what's wanted.f

Thanks, @rachmatowicz for digging into this stuff; it's good to get these contracts in place.

Reply:
Nit: that line is too long. :) It's good to see these contracts without scrolling.

Reply:
done.